uaclient/api/u/pro/attach/auto/full_auto_attach/v1.py

```diff
@@ -0,0 +1,125 @@
+from typing import List, Optional, Tuple
+
+from uaclient import actions, contract, entitlements, messages
+from uaclient.api import exceptions
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.clouds import AutoAttachCloudInstance  # noqa: F401
+from uaclient.config import UAConfig
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.entitlements.entitlement_status import CanEnableFailure
+
+
+class FullAutoAttachOptions(DataObject):
+    fields = [
+        Field("enable", data_list(StringDataValue), False),
+        Field("enable_beta", data_list(StringDataValue), False),
+    ]
+
+    def __init__(
+        self,
+        enable: Optional[List[str]] = None,
+        enable_beta: Optional[List[str]] = None,
+    ):
+        self.enable = enable
+        self.enable_beta = enable_beta
+
+
+class FullAutoAttachResult(DataObject, AdditionalInfo):
+    pass
+
+
+def get_auto_attach_token(cfg: UAConfig, cloud: AutoAttachCloudInstance):
```

Comment:
Since this and `is_any_beta` are helper functions that are not part of the API, let's give them underscores in front of their name to follow the convention to indicate that they're "private".
```suggestion
def _get_auto_attach_token(cfg: UAConfig, cloud: AutoAttachCloudInstance):
```